PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/SplitGraphsTest.java

```diff
@@ -0,0 +1,320 @@
+    nodeIDs.add(3);
+
+    @SuppressWarnings("unchecked") // Necessary when mocking generic types
+        Iterator<Node> mockedIterator = (Iterator<Node>) mock(Iterator.class);
+
+    Node firstNode = mockNode(new String[] {genomeSet.get(0), genomeSet.get(1)}, new Integer[] {3},
+        new Integer[] {3}, 0);
+    Node secondNode = mockNode(new String[] {genomeSet.get(1)}, new Integer[] {0, 1, 3},
+        new Integer[] {0, 1, 3}, 1);
+    Node thirdNode = mockNode(new String[] {genomeSet.get(1), genomeSet.get(2)},
+        new Integer[] {2, 5, 3}, new Integer[] {1, 10, 2}, 3);
+
+    when(mockedIterator.hasNext()).thenReturn(true, true, true, false);
+    when(mockedIterator.next()).thenReturn(firstNode, secondNode, thirdNode);
+    doAnswer(new ForEachAnswer()).when(mockedIterator).forEachRemaining(any());
+    when(mockedGraph.iterator()).thenReturn(mockedIterator);
+    when(mockedGraph.getNode(0)).thenReturn(firstNode);
+    when(mockedGraph.getNode(1)).thenReturn(secondNode);
+    when(mockedGraph.getNode(3)).thenReturn(thirdNode);
+  }
+
+  /**
+   * Creates a mocked node with a specified identifier, genome list, inLinks and outLinks.
+   *
+   * @param genomes  The list of genomes in this node
+   * @param inLinks  The list of inLinks
+   * @param outLinks The list of outLinks
+   * @param identifier       The node ID
+   */
+  private Node mockNode(String[] genomes, Integer[] inLinks, Integer[] outLinks, int identifier) {
+    Node node = mock(Node.class);
+    when(node.getId()).thenReturn(identifier);
+    when(node.getGenomes()).thenReturn(new ArrayList<>(Arrays.asList(genomes)));
+    when(node.getInlinks()).thenReturn(new ArrayList<>(Arrays.asList(inLinks)));
+    when(node.getOutlinks()).thenReturn(new ArrayList<>(Arrays.asList(outLinks)));
+    when(node.getBases()).thenReturn("ACTG");
+    when(node.getAlignment()).thenReturn(1);
+    return node;
+  }
+
+  @Test
+  public void testSplitGraphs() {
+    assertEquals(mockedGraph,
+        AccessPrivate.getFieldValue("graph_field", SplitGraphs.class, defInstance));
+  }
+
+  /**
+   * Verifies that the {@link SplitGraphs#getSubGraph(Collection)} method throws an exception when
+   * invalid genomes are provided.
+   */
+  @Test
+  public void testGetSubGraphThrowsExceptionForInvalidGenomes() {
+    String wrongGenome = "wrong";
+    assertFalse(genomeSet.contains(wrongGenome));
+    ArrayList<String> wrongSet = new ArrayList<>();
+    wrongSet.add(wrongGenome);
+
+    exception.expect(NoSuchElementException.class);
+    defInstance.getSubGraph(wrongSet);
+  }
+
+  /**
+   * Verifies that the {@link SplitGraphs#getSubGraph(Collection)} method returns a non-empty
+   * graph.
+   */
+  @Test
+  public void testGetSubGraphReturnsNonEmptyGraph() {
+    ArrayList<String> genomeSub = new ArrayList<>(
+        Arrays.asList(new String[] {genomeSet.get(0)}));
+
+    OriginalGraph subGraph = defInstance.getSubGraph(genomeSub);
+    assertFalse(subGraph.getNodes().isEmpty());
```

Comment:
You made the `getNodes()` method deprecated in this branch, while in this test file you use this method 11 times. I suggest you create a private static method in this test class which gets the field, using the AccessPrivate class:
```java
  private static HashMap<Integer, Node> getNodes(OriginalGraph graph) {
    return AccessPrivate.getFieldValue("id", OriginalGraph.class, graph);
  }
```